src/protocols/MeterSML.cpp

```diff
@@ -321,9 +328,18 @@ bool MeterSML::_parse(sml_list *entry, Reading *rd) {
 			rd->value(sml_value_to_double(entry->value) * pow(10, scaler));
 		}
 
+		if (_direction_from_status && entry->status) {
+			*direction_from_status = (*(entry->status->data.status16) & (1 << 5)) == (1 << 5);
```

Comment:
should properly check `status->type` before accessing `status->data`.
should replace the magic value "5" with a constant.